### DISCUSSION TITLE
How can we import JS scripts to our page

### DISCUSSION THREAD

**Discussion by shelendravashishtha2** [GrapesJS/grapesjs#4776] [Q&A]
Hi, 
I'm trying to render a template on grapesjs editor, though the template html and css are getting rendered on Editor successfully, Without including JS, The imported template looks like a complete mess, I found Component api in which we can define script but can't find any method so that we can include JS.

I am importing the html and css using below methods

`
editor.setComponents();
editor.setStyle();`

And also if there is any solution to that, is there a way to set html, css and js for pages without even rendering them on editor.

**Comment by rajevar** [comment]
you can insert js/css to the Canvas as follows
`editor.Canvas.getDocument().body.insertAdjacentHTML(
			'afterend',
			'<link rel="stylesheet" type="text/css" href="/style.css" />')`
		);
`

**Comment by artf** [comment]
If you need to import scripts you should enable it first
```js
grapesjs.init({
 // ...
 parser: {
   optionsHtml: { allowScripts: true }
 }
})
```
